### PR TITLE
feat(internal): return registryUrl from datasources

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds `registryUrl` to returned datasource data.

## Context:

This will be useful for future tasks, including knowing if npm presets are fetched from non-default registries or not.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/index.ts [hunk 1]
```diff
@@ -67,6 +67,9 @@ async function getRegistryReleases(
     }
   }
   const res = await datasource.getReleases({ ...config, registryUrl });
+  if (res?.releases.length) {
+    res.registryUrl ||= registryUrl;
```
Comment (viceice) [inline lib/datasource/index.ts:71]:
```suggestion
    res.registryUrl ??= registryUrl;
```

Only assign it `null` or `undefined`, not when `0` or `false` or `''` (empty string)

**Review by viceice — COMMENTED** [review]
just a small thing, LGTM otherwise

File: lib/datasource/types.ts [hunk 2]
```diff
@@ -54,6 +55,7 @@ export interface ReleaseResult {
   homepage?: string;
   sourceUrl?: string;
   sourceDirectory?: string;
+  registryUrl?: string;
```
Comment (viceice) [inline lib/datasource/types.ts:58]: Maybe only add to release?

Comment (rarkins) [reply lib/datasource/types.ts:58]: The result will be cascaded later anyway. I prefer not to put a registry per release for all the ones which don't merge

**Review by viceice — APPROVED** [review]

**Pull request merged**